File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/java/cesar-ch.java
public class Main {
    public static void main(String[] args) {
        // Operadores Aritméticos
        int a = 5;
        int b = 2;
        System.out.println("Suma: " + (a + b));
        System.out.println("Resta: " + (a - b));
        System.out.println("Multiplicación: " + (a * b));
        System.out.println("División: " + (a / (double) b));
        System.out.println("Módulo: " + (a % b));
        System.out.println("Incremento: " + (++a));
        System.out.println("Decremento: " + (--b));

        // Operadores de Comparación
        int x = 10;
        String y = "10";
        System.out.println("Igualdad: " + (x == Integer.parseInt(y)));
        System.out.println("Igualdad estricta: " + (x == Integer.parseInt(y)));
        System.out.println("Desigualdad: " + (a != b));
        System.out.println("Mayor que: " + (a > b));
        System.out.println("Menor que o igual: " + (a <= b));

        // Operadores Lógicos
        boolean p = true;
        boolean q = false;
        System.out.println("AND lógico: " + (p && q));
        System.out.println("OR lógico: " + (p || q));
        System.out.println("NOT lógico: " + (!p));

        // Operadores de Asignación
        int c = 3;
        c += 2;
        System.out.println("Asignación con adición: " + c);

        // Operadores de Identidad
        System.out.println("Identidad: " + (x == 10));
        System.out.println("No identidad: " + (Integer.parseInt(y) != 10));

        // Operadores de Pertenencia
        int[] arreglo = {1, 2, 3};
        System.out.println("Pertenece al arreglo: " + (contains(arreglo, 2)));

        // Operadores de Bits
        int num1 = 5; // Representación binaria: 0101
        int num2 = 3; // Representación binaria: 0011
        System.out.println("AND a nivel de bits: " + (num1 & num2)); // Resultado: 0001 (1 en binario)
        System.out.println("OR a nivel de bits: " + (num1 | num2)); // Resultado: 0111 (7 en binario)
        System.out.println("Desplazamiento a la izquierda: " + (num1 << 1)); // Resultado: 1010 (10 en binario)
        System.out.println("Desplazamiento a la derecha: " + (num1 >> 1)); // Resultado: 0010 (2 en binario)

        // Estructuras de Control
        // Condicionales
        int edad = 18;
        if (18 <= edad && edad <= 125) {
            System.out.println("Eres mayor de edad");
        } else if (edad < 18) {
            System.out.println("Eres menor de edad");
        } else {
            System.out.println("Edad no válida");
        }

        // Excepciones
        try {
            double resultado = 10 / 0;
            System.out.println("Resultado: " + resultado);
        } catch (ArithmeticException e) {
            System.out.println("Error: " + e.getMessage());
        } finally {
            System.out.println("Finalizado el bloque try-catch");
        }

        // Programa
        for (int i = 10; i <= 55; i++) {
            if (i % 2 == 0 && i != 16 && i % 3 != 0) {
                System.out.println(i);
            }
        }
    }

    private static boolean contains(int[] array, int targetValue) {
        for (int value : array) {
            if (value == targetValue) {
                return true;
            }
        }
        return false;
    }
}

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/python/cesar-ch.py
# Operadores Aritméticos
a = 5
b = 2
print("Suma:", a + b)
print("Resta:", a - b)
print("Multiplicación:", a * b)
print("División:", a / b)
print("Módulo:", a % b)
print("Incremento:", a + 1)
print("Decremento:", b - 1)

# Operadores de Comparación
x = 10
y = "10"
print("Igualdad:", x == y)
print("Igualdad ", x == int(y))
print("Desigualdad:", a != b)
print("Mayor que:", a > b)
print("Menor que o igual:", a <= b)

# Operadores Lógicos
p = True
q = False
print("AND lógico:", p and q)
print("OR lógico:", p or q)
print("NOT lógico:", not p)

# Operadores de Asignación
c = 3
c += 2
print("Asignación con adición:", c)

# Operadores de Identidad
print("Identidad:", x == 10)
print("Identidad:", y == 10)

# Operadores de Pertenencia
arreglo = [1, 2, 3]
print("Pertenece al arreglo:", 2 in arreglo)

# Operadores de Bits
num1 = 5  # Representación binaria: 0101
num2 = 3  # Representación binaria: 0011
print("AND a nivel de bits:", num1 & num2)  # Resultado: 0001 (1 en binario)
print("OR a nivel de bits:", num1 | num2)  # Resultado: 0111 (7 en binario)
print("Desplazamiento a la izquierda:", num1 << 1)  # Resultado: 1010 (10 en binario)
print("Desplazamiento a la derecha:", num1 >> 1)  # Resultado: 0010 (2 en binario)

# Estructuras de Control
# Condicionales
edad = 18
if 18 <= edad <= 125:
    print("Eres mayor de edad")
elif edad < 18:
    print("Eres menor de edad")
else:
    print("Edad no válida")

# Excepciones
try:
    resultado = 10 / 0
    print("Resultado:", resultado)
except ZeroDivisionError as error:
    print("Error:", error)
finally:
    print("Finalizado el bloque try-catch")

# Programa
for i in range(10, 56):
    if i % 2 == 0 and i != 16 and i % 3 != 0:
        print(i)